### DISCUSSION TITLE
Feature request: Possibility to use Dialogs with CSS only

### DISCUSSION THREAD

**Discussion by w3teal** [beercss/beercss#404] [Ideas]
I really like that BeerCSS optionalizes javascript, so many elements can be supported with just css. The thing that made me need javascript was `data-ui` to enable the dialog for the navigation drawer.
When i look to CSS-only frameworks like [Picnic CSS](https://picnicss.com/documentation#modal), they have unique way to enable dialog with checkbox hack. I experimenting with Beer CSS to use the same way, and it's just work like charm.
Demo: https://codepen.io/LIGMATV/pen/dPbRjXq

The better part is, we can use the `checkbox` as overlay too, so we can close it by click the outside (overlay). And with `label`, we can easily open and close the dialog anywhere just like what `data-ui` do.
And also, Picnic CSS give a simple javascript code that could enhance the experience allowing to close the modal by pressing `ESC`.
```js
document.onkeydown = function(e){
  if (e.keyCode == 27) {
    var mods = document.querySelectorAll('.dialog-trigger');
    [].forEach.call(mods, function(mod){ mod.checked = false; });
  }
}
```

What do you think by implementing this?

**Comment by laurentpayot** [comment]
@LIGMATV Awesome CSS-only Dialog! It would simplify the UI state management of my app a lot.

The JavaScript code is a bit weird. [`e.keyCode` is deprecated](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode), and the whole thing could be simplified like so:
```js
document.onkeydown = ({ key }) => {
  if (key === 'Escape')
    document.body
      .querySelectorAll('.dialog-trigger')
      .forEach(e => e.checked = false)
}
```
See the <kbd>Escape</kbd> key working with the above JavaScript code at: https://codepen.io/laurentpayot/pen/vEBJMoX

**Reply by w3teal** [reply]
Thank you for forwarding this! I also working for the CSS-only tabs using radio hack (#405), maybe i will make the CSS-only possibility for [snackbars](https://www.beercss.com/#snackbars) using this checkbox approach too. I really want to see the JS is highly optional, so it's can decrease the import request. Maybe JS will used for an situation like this:
> When you have a slider element (we can't update the slider value without JS). And when you have a textarea element (we can't auto resize a textarea without JS). In the future, we will try to leave the JS file fully optional.

**Reply by laurentpayot** [reply]
> I really want to see the JS is highly optional

:pray:

**Comment by leonardorafael** [comment]
I like the CSS hack, but:

Bad:
1) Is not semantic
2) Needs JS to close on ESC button

Good:
1) Multiple dialogs works

I like the popover API too https://codepen.io/leo-bnu/pen/raBpLeB, but:

Bad
1) Multiple dialogs not works

Good:
1) Is semantic
2) Not needs JS to close on ESC button

Maybe we can add the 2 approaches and let the user decide what to use (explaining the limitations). The `data-ui` / `ui` aproaches still have the most complete use-case.

**Reply by laurentpayot** [reply]
Using the popover API would be really awesome, [I’m waiting for it for a year and a half now ](https://github.com/beercss/beercss/discussions/187):wink:
It seems that [on iOS and iPadOS, popovers are not dismissed when the user taps outside of the popover area](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/popover#browser_compatibility). But the issue is [already resolved](https://bugs.webkit.org/show_bug.cgi?id=267688).

By the way, what do you mean by *multiple* dialogs?
From the [MD3 specs](https://m3.material.io/components/dialogs/guidelines#89a9e18e-b7d6-4a8b-9640-07c28a49f8c0):

> A dialog is a modal window that appears in front of app content to provide critical information or ask for a decision. Dialogs disable all app functionality when they appear, and remain on screen until confirmed, dismissed, or a required action has been taken.
Dialogs are purposefully interruptive, so they should be used sparingly. A less disruptive alternative is to use a menu, which provides options without interrupting a user’s experience.

**Reply by leonardorafael** [reply]
> By the way, what do you mean by *multiple* dialogs?
From the [MD3 specs](https://m3.material.io/components/dialogs/guidelines#89a9e18e-b7d6-4a8b-9640-07c28a49f8c0):

I mean at least two opened dialogs. But forget about multiple dialogs. I updated the codepen and it works. There are some strange behavior on close, but it's ok (when clicked oustide, it will close all dialogs and not the last).

I think there are not major problems with popover api now.

**Reply by laurentpayot** [reply]
> I mean at least two opened dialogs. But forget about multiple dialogs. I updated the codepen and it works. 

Oh *stacked* dialogs. That makes sense.

> I think there are not major problems with popover api now.

@leonardorafael JS-less popover dialogs and snackbars (like in the Codepen example) would really make my day. What about menus too?
